### DISCUSSION TITLE
Add `with_x `, `with_y`, to `Vec3`

### DISCUSSION THREAD

**Discussion by silvestrpredko** [bitshifter/glam-rs#404] [Ideas]
I would to discuss the possibility of adding additional API to `Vec3`.

Something like that:

```Rust
#[must_use]
#[inline]
pub fn with_z(mut self, z: f32) -> Self {
    self.z = z;
    self
}
```

For 2d games a lot of time you need to update only one property from `Vec3`. It would be nice to have this out of the box. What do you think? I plan to add this API but at first want to discuss if it will be accepted, and I just don't waste my time.

Thanks

**Comment by bitshifter** [comment]
Seems reasonable, I've created an issue for this #490
